### PR TITLE
mark card button ineligible when using card-fields

### PR DESCRIPTION
**Pull request by wsbrunson**
### Description

the black bcdc button or card button should not be eligible in the smart stack when using the new Hosted Card Fields component (card-fields)

I added a description to explain the purpose of the order of the if statements when checking for eligibility and a test to verify

### CODE REVIEW

File: src/funding/card/config.test.jsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+  },
+  paypal: {
+    instruments: [],
+  },
+  venmo: {
+    instruments: [],
+  },
+});
+
+describe("card eligibility", () => {
```
Comment (elizabethmv) [inline src/funding/card/config.test.jsx:34]: Can we add a test for if `card-fields` is requested and there is a vaulted card as another safe guard for any potential changes to the ordering?

**Review by elizabethmv — APPROVED** [review]
Lgtm! One minor suggestion

File: src/funding/card/config.test.jsx (hunk 1 above)
Comment (wsbrunson) [reply src/funding/card/config.test.jsx:34]: ooooh good call

Comment (wsbrunson) [reply src/funding/card/config.test.jsx:34]: these tests kinda suck, its easy to get them wrong. We need something like Property Testing to really make sure every edge case is covered

**Review by siddy2181 — APPROVED** [review]
Looks good!!
